packages/forklift-console-plugin/src/modules/Providers/utils/validators/common.ts

```diff
@@ -44,7 +44,10 @@ const CERTIFICATE_REGEX = new RegExp(
 const FINGERPRINT_REGEX = /^([a-fA-F0-9]{2}:){19}[a-fA-F0-9]{2}$/;
 
 // validate sub domain names, used in K8s
-const DNS_SUBDOMAINS_NAME_REGEXP = /^[a-z][a-z0-9-]{0,251}[a-z0-9]$/;
+const DNS_LABEL_NAME_REGEXP = '[a-z0-9]([-a-z0-9]*[a-z0-9])?';
+const DNS_SUBDOMAINS_NAME_REGEXP = new RegExp(
+  `^(${DNS_LABEL_NAME_REGEXP}([.]${DNS_LABEL_NAME_REGEXP})*)$`,
+);
```

Comment:
```suggestion
const DNS_SUBDOMAINS_NAME_REGEXP = /^[a-z0-9][a-z0-9-.]{0,251}[a-z0-9]$/;
```

this changes:
a. add numeric char as valid first char
b. add [.] as valid char in the middle of the name.

problems with current solution:
a. removes the length check, ( so it is required it in the validation method  )
b. allow the last char to be . or - , the documentation say "end with an alphanumeric character
"

note: this change in the regexp requires removal of the length check in the method below because it is now done in the regexp, like we do in the rest of the regexp in this file.

Reply:
> The current code that I used was copied from backend forklift validation code and is also relevant to other k8s entities, for example Pod name.

you should follow RFC 1123

the fix I suggest will not block users from entering correct names, but you are correct it can be more restrictive (not sure it's a good thing ... ), here is a more restrictive option: 

```suggestion
const DNS_SUBDOMAINS_NAME_REGEXP = /^(?=.{1,253}$)[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/;
```

this regexp is more complex, but does the job, i don't see the benefit of separating it to label name and subdomain name.

Reply:
I'm ok with the simple (less restrictive) regexp, and more restrictive options too, both will not block users from entring valid names, one is simple but will allow invalid names, the other more restrictive but also more complex.

a comment in the code with refernce to RFC 1123 or link to k8s documentation will be nice touch

Reply:
> The current code that I used was copied from backend forklift validation

I can't find the reference code in forklift, can you add a link to the reference implementation in forklift ?

Reply:
>  i don't see the benefit of separating it to label name and subdomain name.

If there will be no need to validate DNS-LABEL based names then ok. I thought it's more flexible to use the SUBDOMAIN based on the LABEL format, as done on k8s api validation, but sure. We can always add it later on if needed.

>  can't find the reference code in forklift, can you add a link to the reference implementation in forklift ?

The API validation code for  DNS-SUBDOMAIN names (e.g. pod name)  is: https://github.com/kubernetes/apimachinery/blob/0d057e543013c79e20abb000df19bc16d286b777/pkg/util/validation/validation.go#L205

In Forklift we mainly validate based on the DNS-LABEL format for the VM name (e.g. https://github.com/kubev2v/forklift/blob/a1671edf995a93d728c50e90cccff007a6fec3fc/pkg/controller/plan/validation.go#L398) 
No validation for the provider name, but since other k8s entities names uses the DNS-SUBDOMAIN validation then I thought we should do the same.

Reply:
> If there will be no need to validate DNS-LABEL based names then ok. I thought it's more flexible to use the SUBDOMAIN based on the LABEL format, as done on k8s api validation, but sure. We can always add it later on if needed.

Nice :+1:

> The API validation code for DNS-SUBDOMAIN names (e.g. pod name) is: https://github.com/kubernetes/apimachinery/blob/0d057e543013c79e20abb000df19bc16d286b777/pkg/util/validation/validation.go#L205

Greate, we can add a link as remark in our code :fire: 

> If there will be no need to validate DNS-LABEL based names then ok. 

Yes, they have more specific needs, we only need to be more or less accurate, the "real" validation will happen when the API call is made, and then we will show the msg we get.

Lets do something simple (restrictive or less restrictive ) in our end, we can trust the back-end to do the full validation.